### PR TITLE
N* type hints

### PR DESCRIPTION
**Pull request by verailina**
Added type hints in `netapp_cdot_ssh`, `netgear_prosafe_ssh`, `nokia_sros` modules.

### CODE REVIEW

File: netmiko/nokia/nokia_sros.py [hunk 1]
```diff
@@ -152,11 +162,16 @@ def check_config_mode(self, check_string=r"(ex)[", pattern=r"@"):
...
-        return self.send_command(command_string="/admin save", expect_string=r"#")
+        return self._send_command_str(command_string="/admin save", expect_string=r"#")
 
-    def send_config_set(self, config_commands=None, exit_config_mode=None, **kwargs):
+    def send_config_set(  # type: ignore
+        self,
+        config_commands: Union[str, Sequence[str], TextIO, None] = None,
+        exit_config_mode: bool = None,
+        **kwargs: Any,
+    ):
```
Comment (ktbyers) [inline netmiko/nokia/nokia_sros.py:174]: No return value? It should define its return value for the function.

File: netmiko/nokia/nokia_sros.py [hunk 2]
```diff
@@ -244,31 +259,51 @@ class NokiaSrosTelnet(NokiaSros):
...
+        self,
+        ssh_conn: BaseConnection,
+        source_file: str,
+        dest_file: str,
+        file_system: Optional[str] = None,
+        direction: str = "put",
+        socket_timeout: float = 10.0,
+        progress: Optional[Callable[..., Any]] = None,
+        progress4: Optional[Callable[..., Any]] = None,
+        hash_supported: bool = True,
```
Comment (ktbyers) [inline netmiko/nokia/nokia_sros.py:271]: You changed the argument here...needs to be `hash_supported: bool = False`

File: netmiko/nokia/nokia_sros.py [hunk 3]
```diff
@@ -244,31 +259,51 @@ class NokiaSrosTelnet(NokiaSros):
...
-            ssh_conn, source_file, dest_file, hash_supported=hash_supported, **kwargs
+            ssh_conn,
+            source_file,
+            dest_file,
+            file_system,
+            direction,
+            socket_timeout,
+            progress,
+            progress4,
+            hash_supported,
```
Comment (ktbyers) [inline netmiko/nokia/nokia_sros.py:282]: Needs to be named arguments as I wouldn't want to count on the order of arguments here (for anything after the first three arguments).

File: netmiko/nokia/nokia_sros.py [hunk 4]
```diff
@@ -286,8 +321,12 @@ def check_file_exists(self, remote_cmd=""):
                 raise ValueError("Unexpected output from check_file_exists")
         elif self.direction == "get":
             return os.path.exists(self.dest_file)
+        else:
+            raise ValueError("Unexpected value for self.direction")
 
-    def remote_file_size(self, remote_cmd=None, remote_file=None):
+    def remote_file_size(
+        self, remote_cmd: Optional[str] = None, remote_file: Optional[str] = None
```
Comment (ktbyers) [inline netmiko/nokia/nokia_sros.py:328]: I would switch this to be `remote_cmd: str = ""` to make it consistent with what is in `scp_handler` for remote file size.

File: netmiko/nokia/nokia_sros.py [hunk 5]
```diff
@@ -299,11 +338,12 @@ def remote_file_size(self, remote_cmd=None, remote_file=None):
             remote_cmd = self._file_cmd_prefix() + "file dir {}/{}".format(
                 self.file_system, remote_file
             )
-        remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+        remote_out = self.ssh_ctl_chan._send_command_str(remote_cmd)
 
         if "File Not Found" in remote_out:
             raise IOError("Unable to find file on remote system")
 
+        assert remote_file is not None
```
Comment (ktbyers) [inline netmiko/nokia/nokia_sros.py:346]: I would just put an `else` clause up in the self.direction check where if self.direction is neither "put" nor "get" you raise an exception, that should ensure that remote_file is always set to a string and mypy should be able to work it out from that (hopefully).

**Review by ktbyers — CHANGES_REQUESTED** [review]
See comments I posted.

File: netmiko/nokia/nokia_sros.py (hunk 5 above)
Comment (verailina) [reply netmiko/nokia/nokia_sros.py]: Yes, mypy is able to process it :)

**Pull request merged**